tests/ui/allow_attribute.fixed

```diff
@@ -0,0 +1,18 @@
+// run-rustfix
+#![allow(unused)]
+#![warn(clippy::allow_attribute)]
+#![feature(lint_reasons)]
+
+fn main() {}
+
+// Using clippy::needless_borrow just as a placeholder, it isn't relevant.
+
+// Should lint
+#[expect(dead_code)]
+struct T1;
```

Comment:
Could you also add a test with `cfg_attr` I want to see if that work :)

```
// `panic = "unwind"` should always be true
#[cfg_attr(panic = "unwind", allow(dead_code))]
struct CFG_T;
```

I would also like a function with an inner attribute:

```
fn ignore_inner_attr() {
	#![allow(unused)] // Should not lint
}
```

Reply:
Added